pkg/kv/kvserver/testdata/replica_lifecycle/destroy_replica.txt

```diff
@@ -0,0 +1,85 @@
+create-descriptor start=a end=c replicas=(1,2,3)
+----
+created descriptor: r1:{a-c} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=0]
+
+create-replica range-id=1 initialized
+----
+created replica: (n1,s1):1
+Put: 0,0 /Local/RangeID/1/r/RangeLease (0x01698972726c6c2d00): <empty>
+Put: 0,0 /Local/RangeID/1/r/RangeGCThreshold (0x016989726c67632d00): 0,0
+Put: 0,0 /Local/RangeID/1/r/RangeGCHint (0x016989727267636800): latest_range_delete_timestamp:<> gc_timestamp:<> gc_timestamp_next:<> 
+Put: 0,0 /Local/RangeID/1/r/RangeVersion (0x016989727276657200): 10.8-upgrading-step-007
+Put: 0,0 /Local/RangeID/1/r/RangeAppliedState (0x016989727261736b00): raft_applied_index:10 lease_applied_index:10 range_stats:<sys_bytes:142 sys_count:4 > raft_closed_timestamp:<> raft_applied_index_term:5 
```

Comment:
OCD: a bit annoying that the keys here are not ordered. Looking at the `destroy-replica` below, one needs to make some effort to see that everything is deleted.

Feel free to TODO (somewhere in the split trigger code) or ignore.

After #155549, I'm starting to appreciate the idea of generating all batches in keys order. This might have some micro-benefits (I don't know if Pebble works more efficiently with [mostly] ordered batches, but would be interested to know that).